File: PHP/30DaysOfCode/ClassVsInstance.php
<?php
class Person{
    public $age;
    public function __construct($initialAge){
        if($initialAge<0){
            echo "Age is not valid, setting age to 0.\n";
            $this->age = 0;
        } else {
            $this->age = $initialAge;
        }
    }
    public  function amIOld(){
        if($this->age<13){
            $msg="You are young.";
        } else if($this->age<18){
            $msg="You are a teenager.";
        } else {
            $msg="You are old.";
        }
        echo $msg."\n";
    }
    public  function yearPasses(){
        $this->age++;
    }
   
      
}

$T = intval(fgets(STDIN));
for($i=0;$i<$T;$i++){
    $age=intval(fgets(STDIN));
    $p=new Person($age);
    $p->amIOld();
    for($j=0;$j<3;$j++){
        $p->yearPasses();
    }
    $p->amIOld();
    echo "\n";
}

?>

File: JAVA/src/daysOfCode30/ClassVsInstance.java
package daysOfCode30;

import java.util.Scanner;

/**
 *
 * @author Cristian
 */
public class ClassVsInstance {
  
    private int age;	
    public ClassVsInstance(int initialAge) {
        if(initialAge < 0){
            age=0;
            System.out.println("Age is not valid, setting age to 0.");
        } else {
            age=initialAge;
        }
    }

    public void amIOld() {
        String msg = (age<13)?"You are young.":(age<18)?"You are a teenager.":"You are old.";
        System.out.println(msg);
    }

    public void yearPasses() {
        age++;
    }
    
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int i = 0; i < T; i++) {
            int age = sc.nextInt();
            ClassVsInstance p = new ClassVsInstance(age);
            p.amIOld();
            for (int j = 0; j < 3; j++) {
                p.yearPasses();
            }
            p.amIOld();
            System.out.println();
        }
        sc.close();
    }

}
